### PR TITLE
refactor(VCalendar): redo event overlap logic

### LINKED ISSUE #8396: [Bug Report] Tooltips inside day calendar slots don't pop out and reveal whole tooltip
### Environment
**Vuetify Version:** 2.0.5
**Vue Version:** 2.6.10
**Browsers:** Chrome 76.0.3809.87
**OS:** Windows 10

### Steps to reproduce
Hover over any calendar event and see clipped tooltip.

### Expected Behavior
The whole tooltip should be able to be visible.

### Actual Behavior
The tooltip gets clipped because the box for the day has overflow hidden.

### Reproduction Link
<a href="https://codepen.io/sneaky666/pen/JgvGEB?editors=1010" target="_blank">https://codepen.io/sneaky666/pen/JgvGEB?editors=1010</a>

I also need them with "attach" prop because I want it to move as I scroll.

### LINKED ISSUE #8400: [Bug Report] Overlapped events were hidden behind multi-days events in week and day view
### Environment
**Vuetify Version:** 2.0.5
**Vue Version:** 2.6.10
**Browsers:** Chrome 75.0.3770.142
**OS:** Windows 10

### Steps to reproduce
see reproduction link

### Expected Behavior
All events should be visible

### Actual Behavior
The other events were hidden behind the multi-days events

### Reproduction Link
<a href="https://codepen.io/annahoang/pen/zgjBPK?editors=1010" target="_blank">https://codepen.io/annahoang/pen/zgjBPK?editors=1010</a>

### LINKED ISSUE #8565: [Bug Report] Overlapping events on day view causes some event texts to be hidden by other events
### Environment
**Vuetify Version:** 2.0.7
**Vue Version:** 2.6.10
**Browsers:** Firefox 68.0
**OS:** Windows 10

### Steps to reproduce
Create overlapping events on the day view

### Expected Behavior
All event texts should be visible.

### Actual Behavior
On page load, the "Lunch" event texts are hidden by "Another Meeting".

When you click on the event, or go to the next day and back, this fixes itself and shows all the events properly. 

### Reproduction Link
<a href="https://codepen.io/pumpknhd/pen/BaBKqXm?" target="_blank">https://codepen.io/pumpknhd/pen/BaBKqXm?</a>

### Other comments
Instead of doing it the "Google Calendar" way with the overlaps, I think it might be better for the system to create columns as needed.

For example, in the attached CodePen, there are 4 events:
A) Big Meeting - 8-11:30a
B) Meeting - 9-10a
C) Another Meeting - 10-1:30p
D) Lunch - 12-3p

Currently, v-calendar is showing:
A - overlapped by B and C. With C overlapping D.

I'd like to propose that instead of overlapping, it would calculate out 2 columns during that time. So that:

Column 1 |  Column 2
------------ | -------------
Big Meeting | Meeting
Lunch | Another Meeting

### LINKED ISSUE #8620: [Bug Report] Weekly calendar not showing "more-events" button
### Environment
**Vuetify Version:** 2.0.7
**Vue Version:** 2.6.10
**Browsers:** Chrome 76.0.3809.100
**OS:** Mac OS 10.14.4

### Steps to reproduce
Add multiple overlapping events in a weekly calendar

### Expected Behavior
After certain number of events the more-events button should be displayed

### Actual Behavior
Events overlap indefinitely making them unreadable

### Reproduction Link
<a href="https://codepen.io/rayonef/pen/MWgjemz?&editable=true&editors=111" target="_blank">https://codepen.io/rayonef/pen/MWgjemz?&editable=true&editors=111</a>

### LINKED ISSUE #8792: [Bug Report] v-calendar Event is duplicated instead of spanning the days when using event slot
### Environment
**Vuetify Version:** 2.0.11
**Vue Version:** 2.5.17
**Browsers:** Chrome 76.0.3809.132
**OS:** Linux x86_64

### Steps to reproduce
Use the event slot in the calendar.
Add an event that spans multiple days.

### Expected Behavior
The event should span the days instead of duplicating the event

### Actual Behavior
Anything in the event slot essentially is just duplicated as if it were another event instead of spanning the days

### Reproduction Link
<a href="https://codepen.io/kreierson/pen/zYOzJby" target="_blank">https://codepen.io/kreierson/pen/zYOzJby</a>

### LINKED ISSUE #8826: Calendar Events under 1H showing as a Clash when end / start is the same
### Environment
**Vuetify Version:** 2.0.11
**Vue Version:** 2.6.10
**Browsers:** Chrome 76.0.3809.100
**OS:** Mac OS 10.13.6

### Steps to reproduce
You will need to change the event dates start, end to be current date to see the events in the codepen

### Expected Behavior
Events to sit below each other

### Actual Behavior
They sit side by side (like a clash)

### Reproduction Link
<a href="https://codepen.io/nimashoghicompnow/pen/OJLXeQz" target="_blank">https://codepen.io/nimashoghicompnow/pen/OJLXeQz</a>

### Other comments
Only happens for events that are under 1H, next event ending at the same time considered a clash when it actually isn't a clash. 

I have fixed this locally however I cannot compile the project so I can't use it. 
File: vuetify-2.0.1.js 
Line #2862, changed 
FROM: 
'if (other.columnCount === -1 && _this.isSameColumn(visual, other))) {'

TO 
'if (other.columnCount === -1 && _this.isSameColumn(visual, other) && _this.isOverlapping(visual, other)) {'

Hopefully you can update the project so I can download it down with NPM with the fix, so we can use this for a project. I had trouble compiling my own version..

### LINKED ISSUE #8928: [Bug Report] [VCalendar] Event more label can be placed behind or between events
### Environment
**Vuetify Version:** 2.0.16
**Vue Version:** 2.6.10
**Browsers:** Firefox 69.0, Google Chrome
**OS:** Ubuntu undefined

### Steps to reproduce
Add a lot of overlapping multi-day events. In some cases `event-more` label will be placed behind or between event elements.
This happens because the `v-event` element can have `top property` set if they have to be moved to be in line with same event elements in adjacent days.
The fix could be to either set `position: absolute` and `bottom:0` style to event-more (this would fix it to bottom of `v-calendar-weekly__day`), or to add up `top` properties of `VISIBLE`event elements and set it to `event-more` (this would always place it after event elements).

### Expected Behavior
Event-more labels are always displayed below events.

### Actual Behavior
In some cases event-more label is placed behind or between event elements. 

### Other comments
Screenshot of the bug:
event_more_placement_bug

Screenshot after applying `position: absolute` and `bottom:0` style to event-more:
event_more_placement_absolute

Let me know if the position absolute fix is acceptable, I can make the pull request.

### LINKED ISSUE #8937: [Bug Report] [VCalendar] Multi day events overflow container element in week view
### Environment
**Vuetify Version:** 2.0.17
**Vue Version:** 2.6.10
**Browsers:** Firefox 69.0, Google Chrome
**OS:** Ubuntu undefined

### Steps to reproduce
Add overlapping multi day events to calendar, open week view.

### Expected Behavior
All event elements are contained inside `v-calendar-daily_head-day` elements

### Actual Behavior
Some event elements overflow `v-calendar-daily_head-day` elements.

### Reproduction Link
<a href="https://codepen.io/igadmile/pen/YzKLoEg" target="_blank">https://codepen.io/igadmile/pen/YzKLoEg</a>

### Other comments
This happens because some `v-event` elements have set `top` property to be in the same level as `v-event` elements from the same event in adjacent says. The `v-calendar-daily_head-day` takes into account only `v-event` heights for calculating it's own height.

### LINKED ISSUE #9117: [Feature Request] Show week numbers in v-calendar
### Problem to solve
Currently there is no way to see week numbers in v-calendar.

### Proposed solution
A week number option that can be filled out to add week numbers to the calendar view. For example in the leftmost column, above hours.

The config property should be able to specify how week number 1 is found. E.g. the first week with a monday is week 1 or the first 4 day week is week 1.

This was fixed for date-picker, see https://github.com/vuetifyjs/vuetify/issues/5077 for reference.

### LINKED ISSUE #9591: [Bug Report] v-calendar Event Layout : Event is blocked behind other events
### Environment
**Vuetify Version:** 2.1.9
**Vue Version:** 2.6.10
**Browsers:** Chrome 78.0.3904.70
**OS:** Mac OS 10.15.0

### Steps to reproduce
Check reproduction link. The red event is the one that is blocked.

### Expected Behavior
Event should not be blocked behind other events

### Actual Behavior
Event is blocked behind other events

### Reproduction Link
<a href="https://codepen.io/blexxed/pen/wvvmxYE?&editable=true&editors=101" target="_blank">https://codepen.io/blexxed/pen/wvvmxYE?&editable=true&editors=101</a>

### LINKED ISSUE #9786: [Bug Report] Long event names in v-calendar
### Environment
**Vuetify Version:** 2.1.11
**Vue Version:** 2.6.10
**Browsers:** Firefox 70.0
**OS:** Windows 10

### Steps to reproduce
1. Open the link
2. Check the event with `start: "2018-12-31"`
3. The event name is longer than the width of a day tile and gets cut off

### Expected Behavior
If the event spans across multiple day tiles, use all available space to display the event name.

### Actual Behavior
Event name gets cut off

### Reproduction Link
<a href="https://codepen.io/auspex/pen/wvvbNow?&editable=true&editors=101" target="_blank">https://codepen.io/auspex/pen/wvvbNow?&editable=true&editors=101</a>

image

### LINKED ISSUE #9877: [Documentation] Vuetify ---  Calender Weekly not showing on website
### Environment

**Browsers:** Chrome 78.0.3904.108
**OS:** Windows 10

### Steps to reproduce
NA

### Expected Behavior
It should show weekly calender

### Actual Behavior
Weekly calender

### Reproduction Link
<a href="https://vuetifyjs.com/en/components/calendars" target="_blank">https://vuetifyjs.com/en/components/calendars</a>

### PR DESCRIPTION
**Pull request by ClickerMonkey**
- Resolves #8826
- Resolves #8928
- Resolves #8620
- Resolves #8396
- Resolves #8792
- Resolves #8937
- Resolves #8565
- Resolves #8400
- Resolves #9117
- Resolves #9877
- Resolves #9591
- Resolves #9786

## Description
Event overlapping logic was fundamentally flawed and couldn't handle many common scenarios. The calendar now has two available overlap modes: `stack` (imitation of Google Calendar) and `column`. You can also pass your own logic to the `event-overlap-mode` prop.

## How Has This Been Tested?
unit & visually

## Markup:

<details>

```vue
<template>
  <v-sheet class="v-fullscreen">
    <v-toolbar fixed flat app color="white">
      <v-btn outlined @click="setToday" class="mr-3">
        Today
      </v-btn>
      <v-btn fab text small @click="prev">
        <v-icon small>arrow_back_ios</v-icon>
      </v-btn>
      <v-btn fab text small @click="next">
        <v-icon small>arrow_forward_ios</v-icon>
      </v-btn>
      <v-toolbar-title>{{ title }}</v-toolbar-title>
      <v-spacer></v-spacer>
      <v-menu bottom right>
        <template v-slot:activator="{ on }">
          <v-btn outlined class="mx-2" v-on="on">
            <span>{{ typeToLabel[type] }}</span>
            <v-icon right>arrow_drop_down</v-icon>
          </v-btn>
        </template>
        <v-list>
          <v-list-item @click="type = 'day'">
            <v-list-item-title>Day</v-list-item-title>
          </v-list-item>
          <v-list-item @click="type = 'week'">
            <v-list-item-title>Week</v-list-item-title>
          </v-list-item>
          <v-list-item @click="type = 'month'">
            <v-list-item-title>Month</v-list-item-title>
          </v-list-item>
          <v-list-item @click="type = '4day'">
            <v-list-item-title>4 days</v-list-item-title>
          </v-list-item>
        </v-list>
      </v-menu>
      <v-menu bottom right>
        <template v-slot:activator="{ on }">
          <v-btn outlined class="mx-2" v-on="on">
            <span>{{ mode }}</span>
            <v-icon right>arrow_drop_down</v-icon>
          </v-btn>
        </template>
        <v-list>
          <v-list-item @click="mode = 'stack'">
            <v-list-item-title>Stack (default)</v-list-item-title>
          </v-list-item>
          <v-list-item @click="mode = 'column'">
            <v-list-item-title>Column</v-list-item-title>
          </v-list-item>
        </v-list>
      </v-menu>
      <v-switch 
        hide-details
        class="mx-2"
        label="Dark"
        v-model="dark" 
      ></v-switch>
    </v-toolbar>

    <div class="pa-2" style="height: calc(100% - 64px);">

      <v-calendar
        ref="calendar"
        color="primary"
        :events="events"
        :event-color="getEventColor"
        :event-overlap-mode="mode"
        :event-overlap-threshold="45"
        :now="today"
        :type="type"
        :dark="dark"
        v-model="focus"
        @click:event="showEvent"
        @click:more="viewDay"
        @click:date="viewDay"
        @change="updateRange"
      ></v-calendar>

    </div>

    <v-menu
      v-model="selectedOpen"
      :close-on-content-click="false"
      :activator="selectedElement"
      offset-y
    >
      <v-card
        color="grey lighten-4"
        min-width="350px"
        flat
      >
        <v-toolbar
          :color="selectedEvent.color"
          dark
        >
          <v-btn icon>
            <v-icon>edit</v-icon>
          </v-btn>
          <v-toolbar-title v-html="selectedEvent.name"></v-toolbar-title>
          <v-spacer></v-spacer>
          <v-btn icon>
            <v-icon>favorite</v-icon>
          </v-btn>
          <v-btn icon>
            <v-icon>more_vert</v-icon>
          </v-btn>
        </v-toolbar>
        <v-card-content>
          <p class="pl-3" v-html="selectedEvent.details"></p>
          <p class="pl-3">{{ selectedEvent.start }} <strong>(start)</strong></p>
          <p class="pl-3" ng-if="selectedEvent.end">{{ selectedEvent.end }} <strong>(end)</strong></p>
        </v-card-content>
        <v-card-actions>
          <v-btn
            text
            color="secondary"
          >
            Cancel
          </v-btn>
        </v-card-actions>
      </v-card>
    </v-menu>
  </v-sheet>
</template>

<script>
export default {
  data: () => ({
    today: '2019-01-08',
    focus: '2019-01-08',
    dark: false,
    type: 'month',
    mode: 'stack',
    typeToLabel: {
      'month': 'Month',
      'week': 'Week',
      'day': 'Day',
      '4day': '4 Days'
    },
    start: null,
    end: null,
    selectedEvent: {},
    selectedElement: null,
    selectedOpen: false,
    events: [
      {
        name: 'Vacation',
        details: 'Going to the beach!',
        start: '2018-12-29',
        end: '2019-01-01',
        color: 'blue'
      },
      {
        name: 'Meeting',
        details: 'Spending time on how we do not have enough time',
        start: '2019-01-07 09:00',
        end: '2019-01-07 09:45',
        color: 'indigo'
      },
      {
        name: 'Large Event',
        details: 'This starts in the middle of an event and spans over multiple events',
        start: '2018-12-31',
        end: '2019-01-04',
        color: 'deep-purple'
      },
      {
        name: '3rd to 7th',
        details: 'Testing',
        start: '2019-01-03',
        end: '2019-01-07',
        color: 'cyan'
      },
      {
        name: 'Big Meeting',
        details: 'A very important meeting about nothing',
        start: '2019-01-07 08:00',
        end: '2019-01-07 11:30',
        color: 'red'
      },
      {
        name: 'Concurrent Meeting',
        details: 'A very important meeting about nothing concurrently',
        start: '2019-01-07 08:00',
        end: '2019-01-07 09:45',
        color: 'blue'
      },
      {
        name: 'Concurrent Meeting 2',
        details: 'A very important meeting about nothing concurrently',
        start: '2019-01-07 08:00',
        end: '2019-01-07 11:00',
        color: 'blue'
      },
      {
        name: 'Random',
        details: 'A very important meeting about nothing concurrently',
        start: '2019-01-07 10:00',
        end: '2019-01-07 10:45',
        color: 'blue'
      },
      {
        name: 'Another Meeting',
        details: 'Another important meeting about nothing',
        start: '2019-01-07 10:00',
        end: '2019-01-07 13:30',
        color: 'brown'
      },
      {
        name: '7th to 8th',
        start: '2019-01-07',
        end: '2019-01-08',
        color: 'blue'
      },
      {
        name: 'Lunch',
        details: 'Time to feed',
        start: '2019-01-07 12:00',
        end: '2019-01-07 15:00',
        color: 'deep-orange'
      },
      {
        name: 'Nap',
        details: 'Time to rest',
        start: '2019-01-07 15:30',
        end: '2019-01-07 17:00',
        color: 'green'
      },
      {
        name: 'Nap 2',
        details: 'Time to rest',
        start: '2019-01-07 15:30',
        end: '2019-01-07 17:00',
        color: 'green'
      },
      {
        name: 'Nap 3',
        details: 'Time to rest',
        start: '2019-01-07 15:30',
        end: '2019-01-07 17:00',
        color: 'green'
      },
      {
        name: 'Nap 4',
        details: 'Time to rest',
        start: '2019-01-07 16:30',
        end: '2019-01-07 17:00',
        color: 'grey'
      },
      {
        name: 'Nap 5',
        details: 'Time to rest',
        start: '2019-01-07 16:30',
        end: '2019-01-07 17:00',
        color: 'orange'
      },
      {
        name: 'Nap 6',
        details: 'Time to rest',
        start: '2019-01-07 16:30',
        end: '2019-01-07 17:00',
        color: 'black'
      },
      {
        name: '30th Birthday',
        details: 'Celebrate responsibly',
        start: '2019-01-03',
        color: 'teal'
      },
      {
        name: 'New Year',
        details: 'Eat chocolate until you pass out',
        start: '2019-01-01',
        end: '2019-01-02',
        color: 'green'
      },
      {
        name: 'Jan 2nd',
        details: 'The day between two cool days',
        start: '2019-01-02',
        color: 'red'
      },
      {
        name: 'Conference',
        details: 'The best time of my life',
        start: '2019-01-21',
        end: '2019-01-28',
        color: 'grey darken-1'
      },
      {
        name: 'Hackathon',
        details: 'Code like there is no tommorrow',
        start: '2019-01-30 23:00',
        end: '2019-02-01 08:00',
        color: 'black'
      },
      {
        name: 'event 1',
        start: '2019-01-14 18:00',
        end: '2019-01-14 19:00',
        color: '#4285F4'
      },
      {
        name: 'event 2',
        start: '2019-01-14 18:00',
        end: '2019-01-14 19:00',
        color: '#4285F4'
      },
      {
        name: 'event 5',
        start: '2019-01-14 18:00',
        end: '2019-01-14 19:00',
        color: '#4285F4'
      },
      {
        name: 'event 3',
        start: '2019-01-14 18:30',
        end: '2019-01-14 20:30',
        color: '#4285F4'
      },
      {
        name: 'event 4',
        start: '2019-01-14 19:00',
        end: '2019-01-14 20:00',
        color: '#4285F4'
      },
      {
        name: 'event 6',
        start: '2019-01-14 21:00',
        end: '2019-01-14 23:00',
        color: '#4285F4'
      },
      {
        name: 'event 7',
        start: '2019-01-14 22:00',
        end: '2019-01-14 23:00',
        color: '#4285F4'
      },
      {
        name: 'Mash Potatoes 1',
        start: '2019-01-09 12:30',
        end: '2019-01-09 15:30',
        color: '#000000'
      },
      {
        name: 'Mash Potatoes 2',
        start: '2019-01-09 11:30',
        end: '2019-01-09 15:30',
        color: '#000000'
      },
      {
        name: 'Mash Potatoes 3',
        start: '2019-01-09 10:30',
        end: '2019-01-09 15:30',
        color: '#000000'
      },
      {
        name: 'Mash Potatoes 4',
        start: '2019-01-09 11:30',
        end: '2019-01-09 15:30',
        color: '#000000'
      },
      {
        name: '0',
        start: '2019-01-10 00:00',
        end: '2019-01-10 02:00',
        color: '#4285f4'
      },
      {
        name: '1',
        start: '2019-01-10 01:00',
        end: '2019-01-10 03:15',
        color: '#4285f4'
      },
      {
        name: '2',
        start: '2019-01-10 01:45',
        end: '2019-01-10 03:15',
        color: '#4285f4'
      },
      {
        name: '3',
        start: '2019-01-10 01:45',
        end: '2019-01-10 03:00',
        color: '#4285f4'
      },
      {
        name: '4',
        start: '2019-01-10 03:30',
        end: '2019-01-10 08:15',
        color: '#4285f4'
      },
      {
        name: '5',
        start: '2019-01-10 05:00',
        end: '2019-01-10 11:00',
        color: '#4285f4'
      },
      {
        name: '6',
        start: '2019-01-10 05:00',
        end: '2019-01-10 08:15',
        color: '#4285f4'
      },
      {
        name: '7',
        start: '2019-01-10 06:00',
        end: '2019-01-10 09:00',
        color: '#4285f4'
      },
      {
        name: '8',
        start: '2019-01-10 06:00',
        end: '2019-01-10 09:00',
        color: '#4285f4'
      },
      {
        name: '9',
        start: '2019-01-10 08:15',
        end: '2019-01-10 10:45',
        color: '#4285f4'
      },
      {
        name: '10',
        start: '2019-01-10 10:00',
        end: '2019-01-10 15:15',
        color: '#4285f4'
      },
      {
        name: '11',
        start: '2019-01-10 10:00',
        end: '2019-01-10 14:45',
        color: '#4285f4'
      },
      {
        name: '12',
        start: '2019-01-10 10:00',
        end: '2019-01-10 13:15',
        color: '#4285f4'
      },
      {
        name: '13',
        start: '2019-01-10 11:30',
        end: '2019-01-10 15:00',
        color: '#4285f4'
      },

      // 8400
      {
        name: "ALLO",
        start: "2019-01-16 00:00",
        end: "2019-01-16 06:00",
        color: 'black'
      },
      {
        name: "ALLO",
        start: "2019-01-16 01:00",
        end: "2019-01-16 06:00",
        color: 'black'
      },
      {
        name: "DOITT",
        start: "2019-01-16 08:00",
        end: "2019-01-16 16:00",
        color: 'black'
      },
      {
        name: "FNOC",
        start: "2019-01-16 12:00",
        end: "2019-01-17 12:00",
        color: 'black'
      },
      {
        name: "ALLO",
        start: "2019-01-16 21:00",
        end: "2019-01-17 02:00",
        color: 'black'
      },
      {
        name: "ALLO",
        start: "2019-01-17 00:00",
        end: "2019-01-17 06:00",
        color: 'black'
      },
      {
        name: "ALLO",
        start: "2019-01-17 01:00",
        end: "2019-01-17 06:00",
        color: 'black'
      },
      {
        name: "ALLO",
        start: "2019-01-17 01:00",
        end: "2019-01-17 05:00",
        color: 'black'
      },
      {
        name: "ALLO",
        start: "2019-01-17 02:00",
        end: "2019-01-17 03:00",
        color: 'black'
      },
      {
        name: "ALLO",
        start: "2019-01-17 04:00",
        end: "2019-01-17 07:00",
        color: 'black'
      },

      {
        name: '(No title)',
        start: '2019-01-11 02:00',
        end: '2019-01-11 05:00',
        color: '#4285f4'
      },
      {
        name: '(No title)',
        start: '2019-01-11 02:30',
        end: '2019-01-11 05:00',
        color: '#4285f4'
      },
      {
        name: '(No title)',
        start: '2019-01-11 02:30',
        end: '2019-01-11 05:00',
        color: '#4285f4'
      },
      {
        name: '(No title)',
        start: '2019-01-11 03:45',
        end: '2019-01-11 05:45',
        color: '#4285f4'
      },
      {
        name: '(No title)',
        start: '2019-01-11 05:30',
        end: '2019-01-11 09:00',
        color: '#4285f4'
      },
      {
        name: '(No title)',
        start: '2019-01-11 05:30',
        end: '2019-01-11 07:15',
        color: '#4285f4'
      },
      {
        name: '(No title)',
        start: '2019-01-11 06:00',
        end: '2019-01-11 08:00',
        color: '#4285f4'
      },
      {
        name: '(No title)',
        start: '2019-01-11 07:00',
        end: '2019-01-11 08:30',
        color: '#4285f4'
      },
      {
        name: '(No title)',
        start: '2019-01-11 07:00',
        end: '2019-01-11 08:00',
        color: '#4285f4'
      },
      {
        name: '(No title)',
        start: '2019-01-11 07:30',
        end: '2019-01-11 08:30',
        color: '#4285f4'
      },
      {
        name: '(No title)',
        start: '2019-01-11 09:15',
        end: '2019-01-11 11:00',
        color: '#4285f4'
      },
      {
        name: '(No title)',
        start: '2019-01-11 12:00',
        end: '2019-01-11 14:15',
        color: '#4285f4'
      },
      {
        name: '(No title)',
        start: '2019-01-11 13:00',
        end: '2019-01-11 18:45',
        color: '#4285f4'
      },
      {
        name: '(No title)',
        start: '2019-01-11 13:15',
        end: '2019-01-11 14:30',
        color: '#4285f4'
      },
      {
        name: '(No title)',
        start: '2019-01-11 13:30',
        end: '2019-01-11 18:15',
        color: '#4285f4'
      },
      {
        name: '(No title)',
        start: '2019-01-11 17:00',
        end: '2019-01-11 21:15',
        color: '#4285f4'
      },
      {
        name: '(No title)',
        start: '2019-01-11 17:00',
        end: '2019-01-11 20:30',
        color: '#4285f4'
      },
      {
        name: '(No title)',
        start: '2019-01-11 17:30',
        end: '2019-01-11 18:30',
        color: '#4285f4'
      },
      {
        name: '(No title)',
        start: '2019-01-11 18:15',
        end: '2019-01-11 20:15',
        color: '#4285f4'
      },
      {
        name: '(No title)',
        start: '2019-01-11 19:00',
        end: '2019-01-11 20:45',
        color: '#4285f4'
      },
      {
        name: '(No title)',
        start: '2019-01-11 22:00',
        end: '2019-01-11 24:00',
        color: '#4285f4'
      },
      {
        name: '(No title)',
        start: '2019-01-11 22:15',
        end: '2019-01-11 23:15',
        color: '#4285f4'
      },
      {
        name: '(No title)',
        start: '2019-01-11 23:15',
        end: '2019-01-12 00:15',
        color: '#4285f4'
      }
    ]
  }),
  computed: {
    title () {
      const { start, end } = this
      if (!start || !end) {
        return ''
      }

      const startMonth = this.monthFormatter(start)
      const endMonth = this.monthFormatter(end)
      const suffixMonth = startMonth === endMonth ? '' : endMonth

      const startYear = start.year
      const endYear = end.year
      const suffixYear = startYear === endYear ? '' : endYear

      const startDay = start.day + this.nth(start.day)
      const endDay = end.day + this.nth(end.day)

      switch (this.type) {
        case 'month':
          return `${startMonth} ${startYear}`
        case 'week':
        case '4day':
          return `${startMonth} ${startDay} ${startYear} - ${suffixMonth} ${endDay} ${suffixYear}`
        case 'day':
          return `${startMonth} ${startDay} ${startYear}`
      }
      return ''
    },
    monthFormatter () {
      return this.$refs.calendar.getFormatter({
        timeZone: 'UTC', month: 'long'
      })
    }
  },
  mounted () {
    console.log(this.$refs.calendar.days)
  },
  methods: {
    viewDay ({date}) {
      this.focus = date
      this.type = 'day'
    },
    getEventColor (event) {
      return event.color
    },
    setToday () {
      this.focus = this.today
    },
    prev () {
      this.$refs.calendar.prev()
    },
    next () {
      this.$refs.calendar.next()
    },
    showEvent ({ nativeEvent, event }) {
      const open = () => {
        this.selectedEvent = event
        this.selectedElement = nativeEvent.target
        this.selectedOpen = true
      }

      if (this.selectedOpen) {
        this.selectedOpen = false
        setTimeout(open, 10)
      } else {
        open()
      }

      nativeEvent.stopPropagation();
    },
    updateRange ({ start, end }) {
      // You could load events from an outside source (like database) now that we have the start and end dates on the calendar
      this.start = start
      this.end = end
    },
    nth(d) {
      return d > 3 && d < 21
        ? 'th'
        : ['th', 'st', 'nd', 'rd', 'th', 'th', 'th', 'th', 'th', 'th'][d % 10]
    }
  }
}
</script>

<style scoped>
  .v-fullscreen {
    position: absolute;
    padding-top: 1px;
    top: 0px;
    left: 0px;
    right: 0px;
    bottom: 0px;
  }
</style>
```
</details>

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [x] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)
- [ ] I've added new examples to the kitchen (applies to new features and breaking changes in core library)

### CODE REVIEW

File: packages/vuetify/src/components/VCalendar/modes/column.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+export const column: CalendarEventOverlapMode = (events, firstWeekday, overlapThreshold) => {
+  const handler = getOverlapGroupHandler(firstWeekday)
+
+  return (day, dayEvents, timed) => {
+    const visuals = handler.getVisuals(day, dayEvents, timed)
+
+    if (timed) {
+      visuals.forEach(visual => {
+        visual.left = visual.column * 100 / visual.columnCount
```
Comment (sh7dm) [inline packages/vuetify/src/components/VCalendar/modes/column.ts]: what's `100` here?

File: packages/vuetify/src/components/VCalendar/modes/index.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { CalendarEventOverlapMode } from 'types'
+import { stack } from './stack'
+import { column } from './column'
+
+export const CalendarEventOverlapModes: Record<string, CalendarEventOverlapMode> = {
+  stack,
+  column,
+}
```
Comment (sh7dm) [inline packages/vuetify/src/components/VCalendar/modes/index.ts]: Are there more than these two? if not, type can be strenghtened

File: packages/vuetify/src/components/VCalendar/util/timestamp.ts [hunk 3]
```diff
@@ -296,6 +304,16 @@ export function relativeDays (
   return timestamp
 }
 
+export function diffMinutes (min: CalendarTimestamp, max: CalendarTimestamp) {
+  const a = (max.year - min.year) * 525600
+  const b = (max.month - min.month) * 43800
+  const c = (max.day - min.day) * 1440
+  const d = (max.hour - min.hour) * 60
+  const e = (max.minute - min.minute)
```
Comment (sh7dm) [inline packages/vuetify/src/components/VCalendar/util/timestamp.ts]: Lol, maybe create meaningful names for them, like `yearDiff`?

File: packages/vuetify/src/components/VCalendar/mixins/calendar-with-events.ts [hunk 4]
```diff
@@ -201,61 +192,67 @@ export default CalendarBase.extend({
...
+      return this.genEvent(event, scope, false, {
         staticClass: 'v-event',
         class: {
           'v-event-start': start,
           'v-event-end': end,
         },
         style: {
           height: `${eventHeight}px`,
-          top: `${relativeOffset}px`,
+          width: `${width}%`,
```
[the remark this answers is not in the document]
Comment (ClickerMonkey) [reply packages/vuetify/src/components/VCalendar/mixins/calendar-with-events.ts]: These units must be what I've specified, adding convertToUnit not only makes it less clear when reading the code and adds unnecessary overhead. This is also no change to how it was before.

File: packages/vuetify/src/components/VCalendar/modes/index.ts (hunk 2 above)
Comment (ClickerMonkey) [reply packages/vuetify/src/components/VCalendar/modes/index.ts]: Not sure what you mean here. At the moment there are two, but others may be added.

File: packages/vuetify/src/components/VCalendar/mixins/calendar-with-events.ts [hunk 5]
```diff
@@ -201,61 +192,67 @@ export default CalendarBase.extend({
...
 
-      return this.genEvent(event, scope, true, true, {
+      return this.genEvent(event, scope, true, {
         staticClass: 'v-event-timed',
         style: {
           top: `${top}px`,
           height: `${height}px`,
           left: `${left}%`,
-          right: `${right}%`,
+          width: `${width}%`,
```
[the remark this answers is not in the document]
Comment (ClickerMonkey) [reply packages/vuetify/src/components/VCalendar/mixins/calendar-with-events.ts]: These units must be what I've specified, adding convertToUnit not only makes it less clear when reading the code and adds unnecessary overhead. This is also no change to how it was before.

File: packages/vuetify/src/components/VCalendar/VCalendarDaily.ts [hunk 6]
```diff
@@ -65,14 +65,15 @@ export default CalendarWithIntervals.extend({
...
+      const slot = this.$scopedSlots['interval-header']
       const width: string | undefined = convertToUnit(this.intervalWidth)
 
       return this.$createElement('div', {
         staticClass: 'v-calendar-daily__intervals-head',
         style: {
           width,
         },
-      })
+      }, slot ? slot({}) : undefined)
```
Comment (KaelWD) [inline packages/vuetify/src/components/VCalendar/VCalendarDaily.ts]: `getSlot(this, 'interval-header')`

Comment (ClickerMonkey) [reply packages/vuetify/src/components/VCalendar/VCalendarDaily.ts]: Yeah I'll switch to that. That function would be a little more useful if `data?: object` was `getData?: () => object` so if I have slot data it is only created if a slot is actually returned, as opposed to needing to be created and potentially never used.

Comment (KaelWD) [reply packages/vuetify/src/components/VCalendar/VCalendarDaily.ts]: Feel free to add it, `if (typeof data === 'function') data = data()`

Comment (jacekkarczmarczyk) [reply packages/vuetify/src/components/VCalendar/VCalendarDaily.ts]: or `data instanceof Function`, that seems to be the best option https://stackoverflow.com/questions/5999998/check-if-a-variable-is-of-function-type#answer-19717946

**Pull request merged**